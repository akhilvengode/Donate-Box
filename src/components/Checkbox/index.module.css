.checkbox {
  display: inline-flex;
  align-items: center;
  cursor: pointer;
}

.checkbox input {
  opacity: 0;
  width: 0;
  height: 0;
  position: absolute;
}

.checkbox__box,
.checkbox__box--small {
  position: relative;
  box-sizing: border-box;
  display: inline-block;
  background-color: white;
  border: 1px solid lightgrey;
  border-radius: 2px;
  width: 16px;
  height: 16px;
  margin-right: 0.5rem;
}

.checkbox__box:focus {
  outline: 2px solid #007fff;
}

.checkbox__box::after {
  position: absolute;
  box-sizing: content-box;
  left: -2px;
  top: -2px;
  width: 23%;
  height: 80%;
  transform-origin: bottom right;
  transform: rotate(45deg);
  content: "";
  border-width: 0 3px 3px 0;
  border-style: solid;
  border-color: white;
}

.checkbox__box--primary,
.checkbox__box--active {
  background-color: #ea4c89;
  border-color: #ea4c89;
}

.checkbox__box--secondary {
  background-color: #9c27b0;
  border-color: #9c27b0;
}

.checkbox__box--success {
  background-color: #2e7d32;
  border-color: #2e7d32;
}

.checkbox__box--error {
  background-color: #ce0f0f;
  border-color: #ce0f0f;
}

.checkbox__box--medium {
  width: 20px;
  height: 20px;
}

.checkbox__box--large {
  width: 24px;
  height: 24px;
}
