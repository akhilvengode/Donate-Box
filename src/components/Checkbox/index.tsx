import React from "react";
import classnames from "classnames";
import styles from "./index.module.css";

type PropType = {
  size?: "small" | "medium" | "large";
  color?: "primary" | "secondary" | "success";
  label: string;
  checked: boolean;
  onChange: () => void;
};

const Checkbox: React.FC<PropType> = ({
  label,
  onChange,
  checked,
  color = "primary",
  size = "small",
}) => {
  return (
    <label className={styles.checkbox}>
      <span
        tabIndex={0}
        className={classnames(
          styles.checkbox__box,
          styles[`checkbox__box--${size}`],
          checked && styles["checkbox__box--active"],
          checked && styles[`checkbox__box--${color}`]
        )}
      />
      <input
        tabIndex={-1}
        type="checkbox"
        onChange={onChange}
        checked={checked}
      />
      {label}
    </label>
  );
};

export default Checkbox;
