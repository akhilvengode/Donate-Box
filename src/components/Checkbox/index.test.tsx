import { shallow } from "enzyme";
import Checkbox from ".";

describe("Testing checkbox component", () => {
  const onChangeMock = jest.fn();
  const wrapper = shallow(
    <Checkbox label="test-checkbox" onChange={onChangeMock} checked />
  );

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the checkbox", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("should call the event handler function on change event happens", () => {
    wrapper.find("input").simulate("change");
    expect(onChangeMock).toBeCalled();
  });

  it("should be having the active css class when the value of checked prop is true", () => {
    expect(wrapper.find("span").hasClass("checkbox__box--active")).toBeTruthy();
  });

  it("shouldn't be having the active css class when the value of checked prop is false", () => {
    const wrapper = shallow(
      <Checkbox label="test-checkbox" onChange={onChangeMock} checked={false} />
    );
    expect(wrapper.find("span").hasClass("checkbox__box--active")).toBeFalsy();
  });

  it("should be having the size small modifier class for default size", () => {
    expect(wrapper.find("span").hasClass("checkbox__box--small")).toBeTruthy();
  });
});
