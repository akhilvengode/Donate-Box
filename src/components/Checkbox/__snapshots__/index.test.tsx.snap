// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing checkbox component should match snapshot 1`] = `
<label
  className="checkbox"
>
  <span
    className="checkbox__box checkbox__box--small checkbox__box--active checkbox__box--primary"
    tabIndex={0}
  />
  <input
    checked={true}
    onChange={[MockFunction]}
    tabIndex={-1}
    type="checkbox"
  />
  test-checkbox
</label>
`;
